diff --git a/src/daemon/main.c b/src/daemon/main.c
index 6cde209..fba20b9 100644
--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -876,10 +876,15 @@ int main(int argc, char *argv[]) {
 
     pa_memtrap_install();
 
+    #if 0
+ 
+    /* thundersoft remove it will cause pulse audio crash every time. */
+
     if (!getenv("PULSE_NO_SIMD")) {
         pa_cpu_init_x86();
         pa_cpu_init_arm();
     }
+    #endif 
 
     pa_assert_se(mainloop = pa_mainloop_new());
 
